pkg/core/interop_neo.go

```diff
@@ -60,50 +60,75 @@ func storageFind(ic *interop.Context) error {
 	return nil
 }
 
-// createContractStateFromVM pops all contract state elements from the VM
-// evaluation stack, does a lot of checks and returns Contract if it
-// succeeds.
-func createContractStateFromVM(ic *interop.Context) (*state.Contract, error) {
-	script := ic.VM.Estack().Pop().Bytes()
-	if len(script) > MaxContractScriptSize {
-		return nil, errors.New("the script is too big")
+// getNefAndManifestFromVM pops NEF and manifest from the VM's evaluation stack,
+// does a lot of checks and returns deserialized structures if succeeds.
+func getNefAndManifestFromVM(v *vm.VM) (*nef.File, *manifest.Manifest, error) {
+	// Always pop both elements.
+	nefBytes := v.Estack().Pop().BytesOrNil()
+	manifestBytes := v.Estack().Pop().BytesOrNil()
+
+	if err := checkNonEmpty(nefBytes, MaxContractScriptSize); err != nil {
```

Comment:
There's no max length check for NEF in C# node, I think we may left a comment about it. Although NEF deserialisation will fail in case if script is to long.